### PR TITLE
Check auth on initial load

### PR DESCRIPTION
**Pull request by brendansudol**
The main thing this PR does is add a new component `AuthChecker` which wraps all the other components and kicks off a `checkAuth` action that pings `/API/me` to see if auth cookie is there and valid. If so, it loads the content and if not will either redirect to `/login` (if trying to get a private route) or load the public route content.

The other contribution of this PR is a clean-up / modularization of the routes. 

### This pull request is ready to merge when...
- [ ] Tests have been updated (and all tests are passing)
- [ ] This code has been reviewed by someone other than the original author
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented
- ~The change has been documented~
  - ~Associated OpenAPI documentation has been updated~

### This feature is done when...
- ~Design has approved the experience~
- [ ] Product has approved the experience

### CODE REVIEW

**Comment by brendansudol** [comment]
1k deleted lines on the nose!

**Comment by brendansudol** [comment]
spoke too soon!!

File: web/src/containers/AuthChecker.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+AuthChecker.propTypes = {
+  authInit: PropTypes.bool.isRequired,
+  checkAuth: PropTypes.func.isRequired,
+  children: PropTypes.node.isRequired
+};
+
+const mapStateToProps = ({ auth }) => ({ authInit: auth.initialCheck });
+const mapDispatchToProps = { checkAuth };
+
+export default connect(mapStateToProps, mapDispatchToProps)(AuthChecker);
```
Comment (mgwalker) [inline web/src/containers/AuthChecker.js]: So `AuthChecker` just makes sure we've pinged the API before rendering anything, and `TopNav` and `PrivateRoute` do the actual logical checks?  Neat.

**Comment by mgwalker** [comment]
@jeromeleecms @NAretakis So basically what this does, it checks if you're already logged in before showing you the web app.  Because of how we do the authentication, the web app can't know whether you're logged in until it asks the API.  Without the check, the web app has to assume that you're not logged in and makes you do it again.  Now that the check is here, you'll only have to login again when your session expires!

File: web/src/components/Root.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ import PropTypes from 'prop-types';
 import { Provider } from 'react-redux';
 import { ConnectedRouter } from 'react-router-redux';
 
-import App from './App';
+import App from '../components/App';
```
[the remark this answers is not in the document]
Comment (brendansudol) [reply web/src/components/Root.js]: oops, accident

File: web/src/containers/AuthChecker.js (hunk 1 above)
Comment (brendansudol) [reply web/src/containers/AuthChecker.js]: yes, exactly

**Pull request merged**